### ISSUE TITLE
Pre-commit hook misbehaves for UTF-8 files

### ISSUE THREAD

**Issue by jakub-g** [ariatemplates/ariatemplates#196]
When the file to be committed is a "real" UTF-8 file (with or without BOM), i.e. it contains  e.g.  Russian cyryllic characters, then the CRLF replacement in the hook doubles the number of lines, by introducing empty lines.

This seems to be a bug in `dos2unix` bundled with Git.

`dos2unix -D "$FILE"` is meant to convert UNIX -> DOS. It works fine also if the file in already DOS (CRLF) but only for ASCII file. **However, if the file is already DOS && the file is non-ASCII, then it converts CRLF to CRCRLF** thus the output looks like having empty lines introduced every second line.

Using `dos2unix -A` (`A` stands for auto) doesn't introduce this problem. However `dos2unix -A` converts CRLF to LF or LF to CRLF, which is not what we want to achieve. We want to end up with CRLF always, whatever the input is.

**The alternative would be to simply use sed:**
`sed -e 's/$/\r/' input.txt > output.txt`
The above converts LF to CRLF, CRLF to CRLF and works also fine if the input is mixed (LF or CRLF -> CRLF).

**We should however make sure that binary files (i.e. images) will not be corrupted.**

From what I've seen by inspecting `dos2unix -tv` output, being binary or not is decided by having non-ASCII chars, which is bad if we want to have truly Unicode files (because they'll be classified as binary). Otherwise we'll have to stick to \uXXXX in our files.
# TL;DR:
1. The conversion method in the hook should convert LF -> CRLF, CRLF->CRLF, mixed -> CRLF (this is one of the cons of the agreement to have `core.autocrlf=true`)
2. The conversion method should work for UTF-8 files.
3. The hook should leave binary files untouched. It should be stated which files are binary and which not. Probably a whitelist of non-binary files will be created (txt, tpl, css, js...) on which to perform conversion.

**Anyway, we should make sure that binary files (PNGs, GIFs etc.) are unaffected.**

**Probably the best option will be to use `.gitattributes` file to solve all this problems** - to be investigated.
